### PR TITLE
gocryptfs 1.4.3 (new version)

### PR DESCRIPTION
**Pull request by alexanderharm**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

As discussed in https://github.com/Homebrew/homebrew-core/pull/21786 new PR since upstream issue has been fixed.

### CODE REVIEW

File: Formula/gocryptfs.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  version "1.4.3"
+  sha256 "adf55824fd6ca8e258e9207217c52206953293c6ff86e68b58874b6d195e5b11"
+
+  depends_on :osxfuse
+  depends_on "go" => :build
+  depends_on "openssl"
+  depends_on "pkg-config" => :build
+
+  def install
+    mkdir_p "src/github.com/rfjakob/gocryptfs"
```
Comment (DomT4) [inline Formula/gocryptfs.rb]:
```ruby
def install
  ENV["GOPATH"] = buildpath

  (buildpath/"src/github.com/rfjakob/gocryptfs").install buildpath.children
  cd "src/github.com/rfjakob/gocryptfs" do
    system "./build.bash"
    bin.install "gocryptfs"
    prefix.install_metafiles
  end
end
```
Should work?

File: Formula/gocryptfs.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  desc "Encrypted overlay filesystem written in Go"
+  homepage "https://nuetzlich.net/gocryptfs/"
+  url "https://github.com/rfjakob/gocryptfs/releases/download/v1.4.3/gocryptfs_v1.4.3_src-deps.tar.gz"
+  version "1.4.3"
+  sha256 "adf55824fd6ca8e258e9207217c52206953293c6ff86e68b58874b6d195e5b11"
+
+  depends_on :osxfuse
+  depends_on "go" => :build
+  depends_on "openssl"
+  depends_on "pkg-config" => :build
```
Comment (DomT4) [inline Formula/gocryptfs.rb]:
```ruby
depends_on "go" => :build
depends_on "pkg-config" => :build
depends_on "openssl"
depends_on :osxfuse
```

File: Formula/gocryptfs.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    ENV["GOPATH"]=buildpath.to_s
+    ENV["CGO_CFLAGS"]="-I#{Formula["openssl"].opt_include}"
+    ENV["CGO_LDFLAGS"]="-L#{Formula["openssl"].opt_lib}"
+    system "./build.bash"
+
+    bin.install "gocryptfs"
+  end
+
+  test do
+    assertion = "gocryptfs v#{version}"
```
Comment (DomT4) [inline Formula/gocryptfs.rb]: Don't bother version testing if you're also adding a stronger test below.

File: Formula/gocryptfs.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    output = `#{bin}/gocryptfs -version | cut -d ";" -f 1`
+
+    mkdir "cipher"
```
Comment (DomT4) [inline Formula/gocryptfs.rb]:
```ruby
  pipe_output("#{bin}/gocryptfs -init", "password")
  assert_predicate testpath/"gocryptfs.conf"
```
Should work? Unless I'm misunderstanding at a glance.

File: Formula/gocryptfs.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    (buildpath/"src/github.com/rfjakob/gocryptfs").install buildpath.children
+    cd "src/github.com/rfjakob/gocryptfs" do
+      system "./build.bash"
+      bin.install "gocryptfs"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    mkdir testpath/"cipher"
```
Comment (DomT4) [inline Formula/gocryptfs.rb]: It's a nit, but... `(testpath/"cipher").mkpath`.

File: Formula/gocryptfs.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    (buildpath/"src/github.com/rfjakob/gocryptfs").install buildpath.children
+    cd "src/github.com/rfjakob/gocryptfs" do
+      system "./build.bash"
+      bin.install "gocryptfs"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    (testpath/"encdir").mkpath
```
Comment (ilovezfs) [inline Formula/gocryptfs.rb]: do we really need to use a subdir here?

File: Formula/gocryptfs.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    pipe_output("#{bin}/gocryptfs -init #{testpath}/encdir", "password")
```
Comment (ilovezfs) [inline Formula/gocryptfs.rb]: is exit code 0? if so `, 0` for third arg

File: Formula/gocryptfs.rb [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  sha256 "adf55824fd6ca8e258e9207217c52206953293c6ff86e68b58874b6d195e5b11"
+
+  depends_on "go" => :build
+  depends_on "pkg-config" => :build
+  depends_on "openssl"
+  depends_on :osxfuse
+
+  def install
+    ENV["GOPATH"] = buildpath
+
```
Comment (ilovezfs) [inline Formula/gocryptfs.rb]: would probably remove this blank line

File: Formula/gocryptfs.rb [hunk 9]
```diff
@@ -0,0 +1,29 @@
+class Gocryptfs < Formula
+  desc "Encrypted overlay filesystem written in Go"
+  homepage "https://nuetzlich.net/gocryptfs/"
+  url "https://github.com/rfjakob/gocryptfs/releases/download/v1.4.3/gocryptfs_v1.4.3_src-deps.tar.gz"
+  version "1.4.3"
+  sha256 "adf55824fd6ca8e258e9207217c52206953293c6ff86e68b58874b6d195e5b11"
+
+  depends_on "go" => :build
+  depends_on "pkg-config" => :build
+  depends_on "openssl"
```
Comment (ilovezfs) [inline Formula/gocryptfs.rb]: what is the benefit of using this and not `build-without-openssl.bash`?

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/gocryptfs.rb (hunk 9 above)
Comment (alexanderharm) [reply Formula/gocryptfs.rb]: It is a lot faster (4x).

File: Formula/gocryptfs.rb (hunk 6 above)
Comment (alexanderharm) [reply Formula/gocryptfs.rb]: We need an empty dir. Any suggestions?

Comment (DomT4) [reply Formula/gocryptfs.rb]: Spmeone finally gets fed up enough of this being an issue & relocates HOME outside the buildpath/testpath 😸. That’s the actual ideal long-term solution to so many of these, heh.

Comment (alexanderharm) [reply Formula/gocryptfs.rb]: @ilovezfs Can this be merged?

**Review by DomT4 — APPROVED** [review]
This seems okay enough to me if you're happy enough @ilovezfs.

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
Thanks for the :new: formula @alexanderharm! :rocket:

**Pull request closed without merging**